Makefile.one

```diff
@@ -0,0 +1,442 @@
+	@echo ---------------------------------------
+	@echo "STATIC                   $(STATIC)"
+	@echo ---------------------------------------
+	@echo "BPF_VCPU                 $(BPF_VCPU)"
+	@echo "TRACEE_EBPF_OBJ_SRC      $(TRACEE_EBPF_OBJ_SRC)"
+	@echo ---------------------------------------
+	@echo "GO_ARCH                  $(GO_ARCH)"
+	@echo "GO_TAGS_EBPF             $(GO_TAGS_EBPF)"
+	@echo ---------------------------------------
+	@echo "CUSTOM_CGO_CFLAGS        $(CUSTOM_CGO_CFLAGS)"
+	@echo "CUSTOM_CGO_LDFLAGS       $(CUSTOM_CGO_LDFLAGS)"
+	@echo "CGO_EXT_LDFLAGS          $(CGO_EXT_LDFLAGS)"
+	@echo ---------------------------------------
+	@echo "GO_ENV_EBPF              $(GO_ENV_EBPF)"
+	@echo "GO_ENV_RULES             $(GO_ENV_RULES)"
+	@echo ---------------------------------------
+	@echo "TRACEE_EBPF_SRC          $(TRACEE_EBPF_SRC)"
+	@echo ---------------------------------------
+	@echo "GOSIGNATURES_DIR         $(GOSIGNATURES_DIR)"
+	@echo "GOSIGNATURES_SRC         $(GOSIGNATURES_SRC)"
+	@echo ---------------------------------------
+	@echo "REGO_SIGNATURES_DIR      $(REGO_SIGNATURES_DIR)"
+	@echo "REGO_SIGNATURES_SRC      $(REGO_SIGNATURES_SRC)"
+	@echo ---------------------------------------
+
+# usage
+
+.PHONY: help
+help:
+	@echo ""
+	@echo "$ make all                  # build tracee-ebpf, tracee-rules & rules"
+	@echo ""
+	@echo "$ make env                  # show makefile environment/variables"
+	@echo "$ make reqs                 # verify build requirements"
+	@echo ""
+	@echo "# build"
+	@echo ""
+	@echo "$ make core                 # build ./dist/tracee.bpf.core.o"
+	@echo "$ make non-core             # build ./dist/tracee.bpf.XXX.o"
+	@echo "$ make tracee-ebpf          # build ./dist/tracee-ebpf"
+	@echo "$ make tracee-rules         # build ./dist/tracee-rules"
+	@echo "$ make rules                # build ./dist/rules"
+	@echo ""
+	@echo "# clean"
+	@echo ""
+	@echo "$ make clean                # wipe ./dist/"
+	@echo "$ make clean-core           # wipe ./dist/tracee.bpf.core.o"
+	@echo "$ make clean-non-core       # wipe ./dist/tracee.bpf.XXX.o"
+	@echo "$ make clean-tracee-ebpf    # wipe ./dist/tracee-ebpf"
+	@echo "$ make clean-tracee-rules   # wipe ./dist/tracee-rules"
+	@echo "$ make clean-rules          # wipe ./dist/rules"
+	@echo ""
+	@echo "# test"
+	@echo ""
+	@echo "$ make test                 # run all go & opa tests"
+	@echo "$ make tracee-ebpf-test     # go test tracee-ebpf"
+	@echo "$ make tracee-rules-test    # go test tracee-rules"
+	@echo "$ make test-rules           # opa test (tracee-rules)"
+	@echo ""
+
+# variables
+
+BPF_VCPU = v2
+
+# requirements
+
+.PHONY: reqs_cmds
+reqs_cmds: \
+	check_$(CMD_GIT) \
+	check_$(CMD_CLANG) \
+	check_$(CMD_LLC) \
+	check_$(CMD_STRIP) \
+	check_$(CMD_RM) \
+	check_$(CMD_GREP) \
+	check_$(CMD_INSTALL) \
+	check_$(CMD_MKDIR) \
+	check_$(CMD_PKGCONFIG) \
+	check_$(CMD_GO) \
+	check_$(CMD_OPA)
+
+.PHONY: reqs_libs
+reqs_libs: \
```

Comment:
When I run this target in the Vagrant VM I'm getting the following warning:

```
make -f Makefile.one reqs_libs
/usr/lib/x86_64-linux-gnu/pkgconfig/zlib.pc:9: warning: malformed version field with whitespace, trimming to [1.2.11]
/usr/lib/x86_64-linux-gnu/pkgconfig/libelf.pc:8: warning: malformed version field with whitespace, trimming to [0.176]
```

Is it expected?

Reply:
I'll add "--silence-errors" arguments to pkg-config. This error is from your "distro" not versioning libs right (as it seems).